sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/PathsHelper.java

```diff
@@ -578,14 +578,17 @@ public static String generatePathForNameBased(ResourceType resourceType, String
     }
 
     public static String getCollectionPath(String resourceFullName) {
+
+        StringBuilder trimmedResourceFullName = new StringBuilder();
```

Comment:
Why do you use a String Builder? This can all be done on a simple String?

Reply:
You might need an else in the if-block in 587/588 - but I don't quite see why switching to a StringBuilder here?

Reply:
Mostly mutability w/o new heap allocations for `String`. With the if/else block I think allocation count should be same either way I suppose.

Reply:
Fixed in next iteration.